lsp-types/src/Language/LSP/Types/LspId.hs

```diff
@@ -45,3 +46,6 @@ instance Ord SomeLspId where
       go (IdString a) (IdString b) = a `compare` b
       go (IdInt    _) (IdString _) = LT
       go (IdString _) (IdInt    _) = GT
+
+instance Hashable SomeLspId where
+  hashWithSalt n (SomeLspId a) = hashWithSalt n (toBase a)
```

Comment:
I'm a little confused. Why not derive `Hashable` for `LspId` and then just define this in terms of that?

Reply:
Oh sure, that works too

Reply:
I have no idea why it is like this actually, this is old code haha. Sending that change now...